staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -349,8 +369,84 @@ func (o *DebugOptions) generateDebugContainer(pod *corev1.Pod) *corev1.Ephemeral
 	return ec
 }
 
-// waitForEphemeralContainer watches the given pod until the ephemeralContainer is running
-func waitForEphemeralContainer(ctx context.Context, podClient corev1client.PodsGetter, ns, podName, ephemeralContainerName string) (*corev1.Pod, error) {
+// generatePodCopy takes a Pod and return a copy and the debug container name of that copy
+func (o *DebugOptions) generatePodCopyWithDebugContainer(pod *corev1.Pod) (*corev1.Pod, string) {
+	copied := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        o.CopyTo,
+			Namespace:   pod.Namespace,
+			Annotations: pod.Annotations,
+		},
+		Spec: *pod.Spec.DeepCopy(),
+	}
+	copied.Spec.ShareProcessNamespace = pointer.BoolPtr(o.ShareProcesses)
```

Comment:
if `--share-processes` is unset I'd expect the value to be copied from the target pod.

Reply:
Addressed in https://github.com/kubernetes/kubernetes/pull/90094/commits/651503832a6a5cd70e6c019caa06db68490fba23